### PR TITLE
feat(social-media-links): init component - INNO-790

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Refactor of component

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [x] `package.json` is up-to-date and `@ec-europa/ecl-base` is part of the dependencies
- [x] I have checked the dependencies
- [x] I have given the fractal status “ready” to my component
- [x] I have declared `@define mycomponent` in the SCSS file
- [x] I have specified `margin: 0;` on the CSS component
- [x] I have provided tests
- [x] I follow the naming guidelines
- [x] the component supports composition
- [x] there are no hardcoded strings (all content come from the context)
- [x] I have filled the README.md file (at least a few lines)
- [x] My component is listed in the root README
- [x] My PR has the right label(s)

### CODE REVIEW

**Comment by yhuard** [comment]
Deploy preview ready!

Built with commit de50824c887d2ef401bfe99ed580a145b940b792

https://deploy-preview-431--europa-component-library.netlify.com

File: framework/components/ecl-social-icons/ecl-social-icons.twig [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
+      - "variant" (string): variant of the social link. Correspond to the social network (default: empty)
+      - "label" (string): label of the social link (default: empty)
+    }
+    - "extra_classes" (string): extra CSS classes to be added
+    - "extra_attributes" (array): extra attributes classes (optional, format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
 
   2) Variables building a span - this is a default output if the preceding fail.
-  - modifier: (string) Obligatory modifier.
-  - text: (text) The textual value.
+    - "variant" (string): variant of the social link. Correspond to the social network (default: empty)
```
Comment (yhuard) [inline framework/components/ecl-social-icons/ecl-social-icons.twig]: Not used anywhere

File: framework/components/ecl-social-icons/ecl-social-icons.twig [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+{% endif %}
+
+{# Print result #}
+
 {% if link is defined %}
-<a
-  href="{{ link.target }}"
-  class="ecl-link ecl-icon-social {{ link.modifier }} {{ link.extraClasses }}" {{ link.extraAttributes }}
-  >{{ link.title }}</a>
+<a href="{{ link.href }}" class="ecl-link {{ _css_class }}" {{ _extra_attributes }}>{{ link.label }}</a>
```
Comment (yhuard) [inline framework/components/ecl-social-icons/ecl-social-icons.twig]: Why not use `ecl-links`?

File: framework/components/ecl-social-icons/ecl-social-icons.config.js [hunk 3]
```diff
@@ -7,452 +7,451 @@ module.exports = {
...
       <!-- Start demo of: @${item.handle} -->\n
-
       <div>
         <h3>@${item.handle}</h3>
         <div class="ecl-container">
           <div class="ecl-row">
             <div class="ecl-col">
-              <a href="#" class="ecl-link ecl-icon-social ${item.context
-                .modifier}">${item.label}</a>
+              <a href="#" class="ecl-link ecl-icon-social ecl-icon-social--${item
```
Comment (yhuard) [inline framework/components/ecl-social-icons/ecl-social-icons.config.js]: Not consistent with the variants (expects spans, not links)

File: framework/components/ecl-social-media-links/ecl-social-media-links.scss [hunk 4]
```diff
@@ -1,7 +1,50 @@
...
+    margin-left: 0;
+    margin-right: map-get($ecl-spacing, 'm');
+  }
+
+  .ecl-social-media-link__item:last-child {
+    margin-right: 0;
+  }
+}
 
+/*
```
Comment (yhuard) [inline framework/components/ecl-social-media-links/ecl-social-media-links.scss]: Just remove those lines ;)

File: framework/components/ecl-social-media-links/ecl-social-media-links.twig [hunk 5]
```diff
@@ -1,9 +1,77 @@
...
+{% set _css_class = 'ecl-social-media-link' %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+
+{% if variant is defined and variant is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ 'ecl-social-media-link--' ~ variant %}
+{% endif %}
+
+{% if extra_classes is defined %}
```
Comment (yhuard) [inline framework/components/ecl-social-media-links/ecl-social-media-links.twig]: Should test if `is not empty`

File: framework/components/ecl-social-media-links/ecl-social-media-links.twig [hunk 6]
```diff
@@ -1,9 +1,77 @@
...
+{% if extra_attributes is defined %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print result #}
+
+<div class="{{ _css_class }}" {{ _extra_attributes }}>
+  {% block text_block %}
```
Comment (yhuard) [inline framework/components/ecl-social-media-links/ecl-social-media-links.twig]: The logic here could be refined (`has_text` could be replaced by a test on `text`?)

File: framework/components/ecl-social-media-links/ecl-social-media-links.twig [hunk 7]
```diff
@@ -1,9 +1,77 @@
...
+  {% if has_text == true %}
+  <p class="ecl-paragraph">{{ _text }}</p>
+  {% endif %}
+  {% endblock %}
+
+  <ul class="ecl-social-media-link__list ecl-list ecl-list--unstyled{% if variant == 'horizontal' %} ecl-list--inline{% endif %}">
+  {% for _link in links %}
+    <li class="ecl-social-media-link__item">
+      {% include '@ec-europa/ecl-social-icons' with {
+        link: {
```
Comment (yhuard) [inline framework/components/ecl-social-media-links/ecl-social-media-links.twig]: Why not `link: _link` directly?

**Review by yhuard — CHANGES_REQUESTED** [review]
Some points to review

File: framework/components/ecl-social-icons/ecl-social-icons.twig (hunk 1 above)
Comment (emeryro) [reply framework/components/ecl-social-icons/ecl-social-icons.twig]: It is, in the css class

File: framework/components/ecl-social-media-links/ecl-social-media-links.twig (hunk 6 above)
Comment (emeryro) [reply framework/components/ecl-social-media-links/ecl-social-media-links.twig]: That was my first intention, but as text has a default value I didn't know how to test it

**Comment by emeryro** [comment]
Modifications done

File: framework/components/ecl-social-icons/ecl-social-icons.twig [hunk 8]
```diff
@@ -1,25 +1,56 @@
...
+{% endif %}
+
+{# Print result #}
+
 {% if link is defined %}
-<a
-  href="{{ link.target }}"
-  class="ecl-link ecl-icon-social {{ link.modifier }} {{ link.extraClasses }}" {{ link.extraAttributes }}
-  >{{ link.title }}</a>
+{% include '@ec-europa/ecl-links' with {
```
Comment (yhuard) [inline framework/components/ecl-social-icons/ecl-social-icons.twig]: 😍

File: framework/components/ecl-social-media-links/ecl-social-media-links.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+  &:last-child {
+    margin-bottom: 0;
+  }
+}
+
+.ecl-social-media-link__link {
+  align-items: center;
+  display: flex;
+  min-height: 2rem;
```
Comment (yhuard) [inline framework/components/ecl-social-media-links/ecl-social-media-links.scss]: I'm not a huge fan of hardcoded values or `min-height`s, but I'll accept this one

**Review by yhuard — APPROVED** [review]

**Pull request merged**